src/spark_manager.py

```diff
@@ -347,33 +323,182 @@ def _create_master_service(self):
             },
         }
 
+        self._create_or_replace_service(
+            service, self.master_name, "Spark master service"
+        )
+
+    def _create_worker_deployment(
+        self,
+        worker_count: int,
+        worker_cores: int,
+        worker_memory: str,
+        postgres_env: List[Dict[str, str]],
+    ):
+        """
+        Create the Spark worker deployment.
+
+        Args:
+            worker_count: Number of worker replicas
+            worker_cores: CPU cores per worker
+            worker_memory: Memory allocation per worker
+            postgres_env: Postgres environment variables
+        """
+        # Define Spark worker environment variables
+        env = [
+            {"name": "SPARK_MODE", "value": "worker"},
+            {
+                "name": "SPARK_MASTER_URL",
+                "value": f"spark://{self.master_name}.{self.namespace}:{self.DEFAULT_MASTER_PORT}",
+            },
+            {"name": "SPARK_WORKER_CORES", "value": str(worker_cores)},
+            {"name": "SPARK_WORKER_MEMORY", "value": worker_memory},
+            {
+                "name": "SPARK_WORKER_WEBUI_PORT",
+                "value": str(self.DEFAULT_WORKER_WEBUI_PORT),
+            },
+        ] + postgres_env
+
+        # Create the deployment spec
+        deployment = {
+            "apiVersion": "apps/v1",
```

Comment:
Again I don't have enough k8s knowledge to meaningfully review this config. You might want to ask @bio-boris to have a gander

Reply:
seems fine